File: app/src/main/java/com/enipro/data/remote/model/SavedFeed.java
package com.enipro.data.remote.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@org.parceler.Parcel
public class SavedFeed {

    @SerializedName("id")
    @Expose
    public String feedId;

    @SerializedName("created_at")
    @Expose
    public Date createdAt;

    public SavedFeed() {
    }

    public SavedFeed(String feedId) {
        this.feedId = feedId;
    }

    public String getFeedId() {
        return feedId;
    }

    public void setFeedId(String feedId) {
        this.feedId = feedId;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

}

File: app/src/main/java/com/enipro/data/remote/model/SavedFeed.kt
package com.enipro.data.remote.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class SavedFeed(@field:SerializedName("id")
                     @field:Expose var feedId: String) : Parcelable {

    @SerializedName("created_at")
    @Expose
    var createdAt: Date? = null

    constructor(parcel: Parcel) : this(parcel.readString()) {
        createdAt = parcel.readParcelable(Date::class.java.classLoader)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(feedId)
        parcel.writeParcelable(createdAt, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SavedFeed> {
        override fun createFromParcel(parcel: Parcel): SavedFeed {
            return SavedFeed(parcel)
        }

        override fun newArray(size: Int): Array<SavedFeed?> {
            return arrayOfNulls(size)
        }
    }

}
